### PR TITLE
MAINT: stats: Deprecate frechet_l and frechet_r.

### PR DESCRIPTION
**Pull request by WarrenWeckesser**
Fix for https://github.com/scipy/scipy/issues/3258.  This PR is an alternative to https://github.com/scipy/scipy/pull/3275.  The text of the `frechet` distribution docstring was taken from that PR, with only a little editing.  Many thanks to @ligon for getting the work started.

The old names `frechet_l` and `frechet_r`, which were actually what are conventionally called the Weibull distribution, are deprecated.   To deprecate these, the deprecation decorator is applied to each public  method.  The old implementations of `frechet_l` and `frechet_r` are still available as `weibull_max` and `weibull_min`, respectively.

### CODE REVIEW

**Comment by WarrenWeckesser** [comment]
Does anyone know what's going on with the OSX test runs on Travis-CI?

**Comment by rgommers** [comment]
From https://www.traviscistatus.com/:

    Update - We've resumed running private builds at this time. We'll provide further updates
    on the overall progress tomorrow morning PDT. Thank you for your patience.
    Sep 9, 03:13 UTC
    Update - We ran into an issue with booting Xcode 8.x images, so all builds are suspended 
    again. We'll update when private builds are running.
    Sep 9, 01:28 UTC

**Comment by WarrenWeckesser** [comment]
OK, thanks Ralf.

File: scipy/stats/_continuous_distns.py [hunk 1]
```diff
@@ -1640,8 +1634,220 @@ def _munp(self, n, c):
...
+# ['__call__', 'cdf', 'entropy', 'expect', 'fit', 'fit_loc_scale', 'freeze',
+#  'interval', 'isf', 'logcdf', 'logpdf', 'logsf', 'mean', 'median', 'moment',
+#  'nnlf', 'pdf', 'ppf', 'rvs', 'sf', 'stats', 'std', 'var']
+
+_frechet_r_deprec_msg = """\
+The distribution `frechet_r` is a synonym for `weibull_min`; this historical
+usage is deprecated because of possible confusion with the (quite different)
+Frechet distribution.  To preserve the existing behavior of the program, use
+`scipy.stats.weibull_min`.  For the Frechet distribution (i.e. the Type II
+extreme value distribution), use `scipy.stats.frechet`."""
```
Comment (rgommers) [inline scipy/stats/_continuous_distns.py]: Now that you've split the PR, this needs a tweak. ``stats.frechet`` doesn't exist.

Comment (rgommers) [reply scipy/stats/_continuous_distns.py]: Same comment applies to the deprecation message for ``frechet_l``

**Comment by rgommers** [comment]
LGTM except for the minor change needed to the deprecation message.

File: scipy/stats/_continuous_distns.py (hunk 1 above)
Comment (WarrenWeckesser) [reply scipy/stats/_continuous_distns.py]: Oops, missed that one.  I just pushed a fix.

**Comment by rgommers** [comment]
LGTM now, in it goes. Thanks @WarrenWeckesser

**Comment by rgommers** [comment]
@WarrenWeckesser are you going to continue immediately on the other parts that were in gh-7190? If not, then it may be a good idea to open a new issue to keep track of that code.

**Comment by pv** [comment]
Can #3258 be closed now?

**Comment by rgommers** [comment]
> Can #3258 be closed now?

good catch, done

**Pull request merged**